Source/WebKit/Platform/IPC/unix/UnixMessage.h

```diff
@@ -75,15 +87,30 @@ class MessageInfo {
     // The MessageInfo will be copied using memcpy, so all members must be trivially copyable.
     size_t m_bodySize;
     size_t m_attachmentCount;
+#if OS(ANDROID)
+    size_t m_validHardwareBufferCount;
```

Comment:
If anyone is worried about growing this by 8 bytes, we could change the types to `uint8_t`/`uint16_t` because the maximum number of attachments is 254+1, there's the following constants in `ConnectionUnix.cpp`:

```cpp
static const size_t messageMaxSize = 4096;
static const size_t attachmentMaxAmount = 254;
```

I expect the compiler would still add some padding, but in most architectures each member would be 32-bit aligned and the size of the type would be 12 bytes (16 bytes on Android), instead of the current 24 bytes (32 bytes on Android).

Reply:
I wonder why we need this, though. In the case of file descriptors we check the amount of non-null fds on the receiver side, and fill m_fileDescriptors vector with the "valid" ones. We are also checking non-null buffers, so I guess we could do ther same without having to include this info in the message, no? I also wonder if it's possible to send null android buffers.

Reply:
The idea was to avoid yet one more traversal of the array of `AttachmentInfo`s. This is a hot code path after all.

Regarding null buffers, I *think* it is unlikely that we would need to pass them over IPC; but given that the type used everywhere is `RefPtr<AHardwareBuffer>`, which could be null, I decided to handle them in the IPC code.

I am not sure how to proceed. Would you suggest asserting that AHBs cannot be null when they reach the IPC code and simplifying a little bit? (although it will not be much of simplification).